### ISSUE TITLE
Allen and Heath CQ Series

### ISSUE THREAD

**Issue by RossT42** [bitfocus/companion-module-requests#1280]
Hi there could you please look at making a modul for the Allen & Heath CQ12 range of desks .

**Comment by bryce-seifert** [comment]
Do you have any API documentation? I see it has various control apps, but don't see a documented API.

**Comment by AlexRobinsonTV** [comment]
Does this help? See - CQ MixPad 1.1.X

https://www.allen-heath.com/legal/free-and-open-source-software-declarations/

**Comment by bryce-seifert** [comment]
Not quite. It should have information about using remote protocols to control the device, like HTTP, MIDI, OSC, etc and a list of commands to control different actions on the device

**Comment by Timavs** [comment]
Has there been any progress on getting the CQ series module for companion?

**Comment by tehsu** [comment]
MIDI support was released

**Comment by ElijahRietz** [comment]
Here is a link to A&H’s documentation on CQ series MIDI now that it is compatible. https://www.allen-heath.com/content/uploads/2024/06/CQ_MIDI_Protocol_V1_2_0_iss1.pdf

**Comment by RossT42** [comment]
Hi Eijah,

thats great. do u think you guys will build a module for the cq ?

**Comment by ElijahRietz** [comment]
Hey Ross,

I do not have nearly the skills required to build modules, but I’ve reached out to Joseph Adams on Facebook to see if he would be interested in building it.

**Comment by Philip-Jorgensen** [comment]
Can you change the flair of this issue to not have "Missing Documentation" since a MIDI documentation has been released

**Comment by Frenchy2306** [comment]
I would also love to see this module built, I have contacts at A&H and can assist with more documentation as required

**Comment by leonreucher** [comment]
Would anyone be available for testing the module? Because I don't have a unit available. Just reach out on slack please, If somebody is available.

**Comment by leonreucher** [comment]
And also some information regarding the documentation, after talking to A&H:

- Currently it is not possible to control any more functions than described in the documentation. For controlling functions like auto mixing, it needs to be assigned to a Soft Key, which can then be triggered by Companion. It is not clear, if this will ever change.
- Because the dB Values are tapered inside the CQs, only a few values can be set correctly - they can't/won't share the whole alorithm for their dB values.

**Comment by mairginter** [comment]
Hello @leonreucher,
I have a cq-18t in my studio and could test the module from time to time. 

Just let me know if you need it.

**Comment by Timavs** [comment]
> Would anyone be available for testing the module? Because I don't have a unit available. Just reach out on slack please, If somebody is available.

I got you, I’m setting up my cq-20b right now, would love to test the modules

**Comment by leonreucher** [comment]
@Timavs @mairginter would you be so kind to quickly register on the Bitfocus Slack, so we can proceed there?

https://bitfocus.io/api/slackinvite

**Comment by Timavs** [comment]
> @Timavs @mairginter would you be so kind to quickly register on the Bitfocus Slack, so we can proceed there?
> 
> https://bitfocus.io/api/slackinvite

I sent you a pm in slack
